addons/account/models/chart_template.py

```diff
@@ -128,345 +130,383 @@ class AccountChartTemplate(models.Model):
-            # entries created for the company.
-            if self.existing_accounting(company):
-                raise UserError(_('Could not install new chart of account as there are already accounting entries existing.'))
-
-            # delete accounting properties
-            prop_values = ['account.account,%s' % (account_id,) for account_id in existing_accounts.ids]
-            existing_journals = self.env['account.journal'].search([('company_id', '=', company.id)])
-            if existing_journals:
-                prop_values.extend(['account.journal,%s' % (journal_id,) for journal_id in existing_journals.ids])
-            self.env['ir.property'].sudo().search(
-                [('value_reference', 'in', prop_values)]
-            ).unlink()
-
-            # delete account, journal, tax, fiscal position and reconciliation model
-            models_to_delete = ['account.reconcile.model', 'account.fiscal.position', 'account.tax', 'account.move', 'account.journal', 'account.group']
-            for model in models_to_delete:
-                res = self.env[model].sudo().search([('company_id', '=', company.id)])
-                if len(res):
-                    res.unlink()
-            existing_accounts.unlink()
-
-        company.write({'currency_id': self.currency_id.id,
-                       'anglo_saxon_accounting': self.use_anglo_saxon,
-                       'bank_account_code_prefix': self.bank_account_code_prefix,
-                       'cash_account_code_prefix': self.cash_account_code_prefix,
-                       'transfer_account_code_prefix': self.transfer_account_code_prefix,
-                       'chart_template_id': self.id
-        })
+        models_to_cleanup = [
+            'account.reconcile.model',
+            'account.fiscal.position',
+            'account.tax',
+            'account.move',
+            'account.journal',
+            'account.group',
+        ]
+        property_to_cleanup_domains = []
+        for model in models_to_cleanup:
+            self.env[model].sudo().search([('company_id', '=', company.id)]).unlink()
+            property_to_cleanup_domains.append([('value_reference', 'like', 'account.journal,%')])
 
-        #set the coa currency to active
-        self.currency_id.write({'active': True})
+        self.env['ir.property'].sudo()\
+            .search([('company_id', '=', company.id)] + expression.OR(property_to_cleanup_domains))\
+            .unlink()
 
-        # When we install the CoA of first company, set the currency to price types and pricelists
-        if company.id == 1:
-            for reference in ['product.list_price', 'product.standard_price', 'product.list0']:
-                try:
-                    tmp2 = self.env.ref(reference).write({'currency_id': self.currency_id.id})
-                except ValueError:
-                    pass
+    @api.model
+    def _prepare_payment_acquirer_account(self, company, **kwargs):
```

Comment:
A function `_prepare_accounts` returning an empty mapping `key -> create_vals` inherited in the module `payment` would be cleaner IMO. It will also be possible to create dynamic accounts from l10n this way if needed. 

However, I feel like this is only call in post init hooks? So if you create a company after having installed various `payment_*` modules, this will not be created?

And why `kwargs`? You are hardcoding values of kwargs in the extensions but not using it here? Seems really weird...